internal/driver/gomobile/canvas.go

```diff
@@ -451,6 +474,28 @@ func (c *mobileCanvas) tapUp(pos fyne.Position, tapID int,
 	}
 }
 
+func (c *mobileCanvas) waitForDoubleTap(co fyne.CanvasObject, ev *fyne.PointEvent, tapCallback func(fyne.Tappable, *fyne.PointEvent), doubleTapCallback func(fyne.DoubleTappable, *fyne.PointEvent)) {
+	var ctx context.Context
+	ctx, c.touchCancelFunc = context.WithDeadline(context.TODO(), time.Now().Add(time.Millisecond*doubleClickDelay))
+	defer c.touchCancelFunc()
+	select {
+	case <-ctx.Done():
+		if c.touchTapCount == 2 && c.touchLastTapped == co {
+			if wid, ok := co.(fyne.DoubleTappable); ok {
+				doubleTapCallback(wid, ev)
+			}
+		} else {
+			if wid, ok := co.(fyne.Tappable); ok {
+				tapCallback(wid, ev)
+			}
+		}
+		c.touchTapCount = 0
+		c.touchCancelFunc = nil
+		c.touchLastTapped = nil
+		return
+	}
```

Comment:
nit: select unnecessary
```suggestion
	<-ctx.Done():
	if c.touchTapCount == 2 && c.touchLastTapped == co {
		if wid, ok := co.(fyne.DoubleTappable); ok {
			doubleTapCallback(wid, ev)
		}
	} else {
		if wid, ok := co.(fyne.Tappable); ok {
			tapCallback(wid, ev)
		}
	}
	c.touchTapCount = 0
	c.touchCancelFunc = nil
	c.touchLastTapped = nil
	return
```

Reply:
resolved